src/torch_models.py

```diff
@@ -437,6 +435,30 @@ def _fit(self, inputs: Array, outputs: Array) -> None:
         print(f"Loaded best model with loss: {loss:.5f}")
 
 
+class MLPClassifierEnsemble(Classifier):
+    """MLPClassifierEnsemble definition."""
+
+    def __init__(self, in_size: int, max_itr: int, n: int) -> None:
+        self._members = [
+            MLPClassifier(in_size, max_itr, CFG.seed + i) for i in range(n)
+        ]
+
+    def fit(self, X: Array, y: Array) -> None:
+        for i, member in enumerate(self._members):
+            print(f"Fitting member {i} of ensemble...")
+            member.fit(X, y)
+
+    def classify(self, x: Array) -> bool:
+        member_vals = []
+        for member in self._members:
+            x_normalized = member._normalize(x)  # pylint: disable=protected-access
+            member_vals.append(member(x_normalized).item())
+        avg = np.mean(member_vals)
```

Comment:
it probably won't make a difference here, because I'm guessing most outputs are extremely close to 0 or 1, but I'm wondering whether it's preferable to aggregate the ensemble outputs before the `> 0.5`, or after. can you look for some reputable source / other code that uses ensembles for classification and see what they do? maybe scikit learn?

Reply:
from what i can tell in the scikit source code, they have both implementations
![image](https://user-images.githubusercontent.com/26547725/155229112-a31355d0-a693-45fe-8855-d561b6d0474c.png)
aggregating before is what it does if the individual members are probabilistic, while aggregating after is called "voting". also while i was trying to figure out why the ensemble wasn't working, i printed out some values and surprisingly, not all of them were always very close to 0 or 1 (i saw like a 0.7), so i would like to keep it like this?

Reply:
huh good to know! thanks for looking into it, sounds good.